testsupport/wait/awaitility.go

```diff
@@ -161,16 +156,16 @@ func (a *Awaitility) WaitForService(t *testing.T, name string) (corev1.Service,
 // WaitForToolchainClusterWithCondition waits until there is a ToolchainCluster representing a operator of the given type
 // and running in the given expected namespace. If the given condition is not nil, then it also checks
 // if the CR has the ClusterCondition
-func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespace string, condition *toolchainv1alpha1.ToolchainClusterCondition) (toolchainv1alpha1.ToolchainCluster, error) {
+func (a *Awaitility) WaitForToolchainClusterWithCondition(t *testing.T, namespace string, cdtype toolchainv1alpha1.ConditionType) (toolchainv1alpha1.ToolchainCluster, error) {
 	t.Logf("waiting for ToolchainCluster in namespace '%s'", namespace)
 	timeout := a.Timeout
-	if condition != nil {
+	if cdtype != "" {
 		timeout = ToolchainClusterConditionTimeout
 	}
 	var c toolchainv1alpha1.ToolchainCluster
 	err := wait.Poll(a.RetryInterval, timeout, func() (done bool, err error) {
```

Comment:
The function is not actually called without the ConditionType parameter specified, so this if statement doesn't make much sense anymore. 
```suggestion
	var c toolchainv1alpha1.ToolchainCluster
	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
```

Reply:
yea removed it now
Had thought about it, but wasnt sure about why we were setting a different timeout and hence had left it,